src/internal/operations/backup_integration_test.go

```diff
@@ -1087,6 +1099,371 @@ func (suite *BackupOpIntegrationSuite) TestBackup_Run_oneDrive() {
+
+	fileDBF := func(id, timeStamp, subject, body string) []byte {
+		return []byte(id + subject)
+	}
+
+	// Populate initial test data.
+	// Generate 2 new folders with two items each. Only the first two
+	// folders will be part of the initial backup and
+	// incrementals. The third folder will be introduced partway
+	// through the changes. This should be enough to cover most delta
+	// actions.
+	for _, destName := range genDests {
+		generateContainerOfItems(
+			t,
+			ctx,
+			gc,
+			path.OneDriveService,
+			acct,
+			path.FilesCategory,
+			selectors.NewOneDriveRestore(owners).Selector,
+			m365.AzureTenantID, suite.user, driveID, destName,
+			2,
+			// Use an old backup version so we don't need metadata files.
+			0,
+			fileDBF)
+	}
+
+	containerIDs := map[string]string{}
+
+	// verify test data was populated, and track it for comparisons
+	for _, destName := range genDests {
+		// Use path-based indexing to get the folder's ID. This is sourced from the
+		// onedrive package `getFolder` function.
+		itemURL := fmt.Sprintf(
+			"https://graph.microsoft.com/v1.0/drives/%s/root:/%s",
+			driveID,
+			destName)
+		resp, err := msdrive.NewItemsDriveItemItemRequestBuilder(itemURL, gc.Service.Adapter()).
+			Get(ctx, nil)
+		require.NoErrorf(t, err, "getting drive folder ID", "folder name: %s", destName)
+
+		containerIDs[destName] = ptr.Val(resp.GetId())
+	}
+
+	// container3 does not exist yet. It will get created later on
+	// during the tests.
+	containers := []string{container1, container2, container3}
+	sel := selectors.NewOneDriveBackup(owners)
+	sel.Include(
+		sel.Folders(containers, selectors.PrefixMatch()),
+	)
+
+	bo, _, kw, ms, closer := prepNewTestBackupOp(t, ctx, mb, sel.Selector, ffs)
+	defer closer()
+
+	// run the initial backup
+	runAndCheckBackup(t, ctx, &bo, mb)
+
+	var (
+		newFile     models.DriveItemable
+		newFileName = "new_file.txt"
+	)
+
+	// Although established as a table, these tests are no isolated from each other.
+	// Assume that every test's side effects cascade to all following test cases.
+	// The changes are split across the table so that we can monitor the deltas
+	// in isolation, rather than debugging one change from the rest of a series.
+	table := []struct {
+		name string
+		// performs the incremental update required for the test.
+		updateUserData func(t *testing.T)
+		itemsRead      int
+		itemsWritten   int
+	}{
+		{
+			name:           "clean incremental, no changes",
+			updateUserData: func(t *testing.T) {},
+			itemsRead:      0,
+			itemsWritten:   0,
+		},
+		{
+			name: "create a new file",
+			updateUserData: func(t *testing.T) {
+				targetContainer := containerIDs[container1]
+				driveItem := models.NewDriveItem()
+				driveItem.SetName(&newFileName)
+				driveItem.SetFile(models.NewFile())
+				newFile, err = onedrive.CreateItem(
+					ctx,
+					gc.Service,
+					driveID,
+					targetContainer,
+					driveItem,
+				)
```

Comment:
nit, outlier on other consistency
```suggestion
					driveItem)
				
```